File: client/src/modules/staff/common/services/staffWorkspaceService.js
import axiosInstance from '@/core/api/axios';

const normalizeBase = (base) => base.replace(/\/$/, '');

const staffWorkspaceService = {
  async getDashboard(apiBase) {
    const { data } = await axiosInstance.get(`${normalizeBase(apiBase)}/dashboard/`);
    return data;
  },

  async getProfile(apiBase) {
    const { data } = await axiosInstance.get(`${normalizeBase(apiBase)}/profile/`);
    return data;
  },

  async getItems(apiBase, endpoint) {
    const { data } = await axiosInstance.get(`${normalizeBase(apiBase)}/${endpoint}/`);
    return data;
  },

  async changePassword(apiBase, payload) {
    const { data } = await axiosInstance.post(
      `${normalizeBase(apiBase)}/change-password/`,
      payload,
    );
    return data;
  },
};

export default staffWorkspaceService;

File: server/apps/staff/services/staff_workspace_service.py
class StaffWorkspaceService:
    @staticmethod
    def get_profile(user, profile_attr, role_label):
        profile = getattr(user, profile_attr, None)
        if profile is None:
            raise ValueError(f"Only {role_label} staff can access this endpoint.")
        return profile

    @staticmethod
    def update_profile(user, profile_attr, role_label, data):
        profile = StaffWorkspaceService.get_profile(user, profile_attr, role_label)
        allowed_fields = ["work_phone", "office_location", "notes"]

        for field in allowed_fields:
            if field in data:
                setattr(profile, field, data[field])

        profile.save()
        return profile

    @staticmethod
    def dashboard(user, profile_attr, role_label, default_work):
        profile = StaffWorkspaceService.get_profile(user, profile_attr, role_label)
        permissions = list(
            profile.permissions.filter(is_active=True).values_list("code", flat=True)
        )
        return {
            "profile": {
                "id": str(profile.id),
                "full_name": user.full_name,
                "email": user.email,
                "firm_role": profile.firm_role,
                "job_title": profile.job_title,
                "law_firm": profile.law_firm.name,
            },
            "summary": {
                "active_permissions": len(permissions),
                "pending_tasks": 0,
                "documents": 0,
                "notifications": 0,
            },
            "permissions": permissions,
            "default_work": default_work(profile),
            "recent_activity": [
                {
                    "id": "activity-001",
                    "title": f"{role_label} dashboard ready",
                    "description": "Your workspace is active.",
                }
            ],
        }

    @staticmethod
    def change_password(user, profile_attr, role_label, *, old_password, new_password):
        StaffWorkspaceService.get_profile(user, profile_attr, role_label)

        if not user.check_password(old_password):
            raise ValueError("Old password is incorrect.")

        user.set_password(new_password)
        user.must_change_password = False
        user.save(update_fields=["password", "must_change_password"])
        return True

    @staticmethod
    def placeholder_items(user, profile_attr, role_label, item_type):
        StaffWorkspaceService.get_profile(user, profile_attr, role_label)
        return [
            {
                "id": f"{item_type}-001",
                "title": f"{role_label} {item_type.replace('-', ' ')} workspace ready",
                "status": "READY",
            }
        ]
